tests/integration/test_int.py

```diff
@@ -0,0 +1,454 @@
+            "test name",
+            "--branch",
+            "test",
+            "--dry-run",
+        ],
+    )
+    # Check the CLI sync-cac-content is successful
+    print("Current directory: {}".format(pathlib.Path(os.curdir).resolve()))
+    out = subprocess.run(["find", ".", "-ls"], capture_output=True, text=True)
+    print(result.stdout)
+    # print(result.stderr)
+    print(result.exit_code)
+    assert result.exit_code == 0
+    component_definition = repo_path.joinpath(test_comp_path)
+    # Check if the component definition is created
+    assert component_definition.exists()
+
+    # Fix trestle:// to file://
+
+    new_cat_json_text = (
+        pathlib.Path(repo_dir) / "catalogs/simplified_nist_catalog/catalog.json"
+    ).read_text()
+    new_cat_json_text = new_cat_json_text.replace(
+        "trestle://catalogs/simplified_nist_catalog/catalog.json",
+        "file://controls/catalog.json",
+    )
+    new_cat_json = json.loads(new_cat_json_text)
+
+    new_prof_json_text = (
+        pathlib.Path(repo_dir) / "profiles/simplified_nist_profile/profile.json"
+    ).read_text()
+    new_prof_json_text = new_prof_json_text.replace(
+        "trestle://catalogs/simplified_nist_catalog/catalog.json",
+        "file://controls/catalog.json",
+    )
+    new_prof_json_text = new_prof_json_text.replace(
+        '"param_id"', '"param-id"'
+    )  # TODO compliance-trestle uses param_id and param-id interchangably, complytime requires param-id
+    new_prof_json = json.loads(new_prof_json_text)
+
+    new_cd_json_text = component_definition.read_text()
+    new_cd_json_text = new_cd_json_text.replace(
+        "trestle://profiles/simplified_nist_profile/profile.json",
+        "file://controls/profile.json",
+    )
+    new_cd_json = json.loads(new_cd_json_text)
+
+    with open(
+        (complytime_home / ".config/complytime/controls/catalog.json"), "w"
+    ) as file:
+        json.dump(new_cat_json, file)
+    with open(
+        (complytime_home / ".config/complytime/controls/profile.json"), "w"
+    ) as file:
+        json.dump(new_prof_json, file)
+    with open(
+        (complytime_home / ".config/complytime/bundles/component-definition.json"), "w"
+    ) as file:
+        json.dump(new_cd_json, file)
+
+    # shutil.copy(component_definition, complytime_home / '.config/complytime/bundles/')
+    # shutil.copy(pathlib.Path(repo_dir) / 'catalogs/simplified_nist_catalog/catalog.json',
+    #   complytime_home / '.config/complytime/bundles/')
+    # shutil.copy(pathlib.Path(repo_dir) / 'profiles/simplified_nist_profile/profile.json',
+    #   complytime_home / '.config/complytime/controls/')
+
+    result = subprocess.run(
+        ["complytime", "list", "--plain"],
+        # cwd=complytime_home,
+        capture_output=True,
+    )
+    assert result.returncode == 0
+    assert b"Title" in result.stdout
+    assert b"Framework ID" in result.stdout
+
+
+@pytest.mark.slow
+def test_compdef_type_software_sync(
```

Comment:
This test is pretty similar to the previous one `test_full_sync` and the next one `test_compdef_type_validation_sync`. What about creating a common function used by two tests in order to avoid code duplication? It would be also easier to maintain.

Reply:
Summary of our chat about code duplication in unit tests vs. production code: I sometimes prefer to have duplication in unit tests so that the tests can be self-contained within a function